### DISCUSSION TITLE
How plumed compute force from cv?

### DISCUSSION THREAD

**Discussion by liangzhixin-202169** [plumed/plumed2#1094] [Q&A]
if I design a new cv, I wonder how plumed compute its derivatives. Is there an automatic differentiation framework, or should I manually derive the formula？

**Comment by GiovanniBussi** [comment]
Hi. What do you mean for "design"?

If you want to write a new CV in C++, you need to code the derivatives as well.

If you build it combining existing variables (e.g. with COMBINE, CUSTOM, etc), then plumed will know the derivative of each building block and do the force propagation.

I hope this helps
